test-scripts/integration_tests.py

```diff
@@ -0,0 +1,166 @@
+        self.__lostromos_process = subprocess.Popen(
+            [
+                _LOSTROMOS_EXE,
+                "start",
+                "--config",
+                _LOSTROMOS_CONFIGURATION_FILE,
+            ],
+            stdout=subprocess.PIPE
+        )
+        print("Started Lostromos with PID: {}".format(self.__lostromos_process.pid))
+
+        self.__wait_for_lostromos_start()
+        self.__check_metrics(2, 2, 2, 0, 0)
+        self.__kubectl.apply(_NEMO_CUSTOM_RESOURCE_FILE)
+        self.__check_metrics(3, 3, 3, 0, 0)
+        self.__kubectl.apply(_NEMO_UPDATE_CUSTOM_RESOURCE_FILE)
+        self.__check_metrics(4, 3, 3, 0, 1)
+        self.__kubectl.delete(_THINGS_CUSTOM_RESOURCE_FILE)
+        self.__check_metrics(6, 1, 3, 2, 1)
+        self.__kubectl.delete(_NEMO_CUSTOM_RESOURCE_FILE)
+        self.__check_metrics(7, 0, 3, 3, 1)
+
+    def tearDown(self):
+        """
+        Kill the lostromos process if it was created.
+        """
+        if self.__lostromos_process:
+            self.__lostromos_process.kill()
+
+    def __check_metrics(self, events, managed, created, deleted, updated):
+        """
+        Check the metrics output to ensure that what we are expecting has occurred. Will wait up to 10 seconds looking
+        for the expected amount of events to have occurred. If the events haven't occurred, then an assertionError will
+        be raised. If the events occurred, we will check the stats for the managed/created/deleted/updated resources.
+        :param events: Number of events we are expecting to have happened.
+        :param managed: Number of resources we expect Lostromos to be managing.
+        :param created: Number of resources we expect Lostromos to have created.
+        :param deleted: Number of resources we expect Lostromos to have deleted.
+        :param updated: Number of resources we expect Lostromos to have updated.
+        """
+        metrics = []
+        attempts = 10
+        while attempts > 0:
+            metrics_response = requests.get(self.__metrics_url)
+            metrics_response.raise_for_status()
+            metrics = metrics_response.text.split("\n")
+            if "releases_total_events {}".format(events) not in metrics:
+                sleep(1)
+                attempts -= 1
+            else:
+                self.assertIn("releases_managed_total {}".format(managed), metrics)
+                self.assertIn("releases_created_count {}".format(created), metrics)
+                self.assertIn("releases_deleted_count {}".format(deleted), metrics)
+                self.assertIn("releases_updated_count {}".format(updated), metrics)
+                return
+        raise AssertionError("Failed to see the expected number of events. {}".format(metrics))
+
+    def __wait_for_lostromos_start(self):
+        """
+        Wait for Lostromos to start up, then return.
+        """
+        seconds_to_wait = 60
+        seconds_to_sleep = 5
```

Comment:
Seems a little long, no harm in checking every second.

Reply:
I was initially doing go run main.go ..., which caused the time inflation, but you are correct in that now that I'm using the already build lostromos binary it shouldn't take more than 5-10 seconds at most. Giving 15 just because this is going to run in an environment we don't control, and could be hit by slow downs we don't expect.